graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,99 @@
+# Alternate Key Pattern
+
+Microsoft Graph API Design Pattern
+
+_The Alternate Key Pattern provides the ability to query for a single, specific resource identifiable through an alternative property (called key) that is not its unique identifier_
+
+## Problem
+
+---
+
+The resources exposed in Graph are identified by an [UUID (Universally Unique Identifier)](https://en.wikipedia.org/wiki/Universally_unique_identifier) - which guarantees uniqueness inside the same resource type. Often though, that same resource can also be uniquely identified by an alternative, more convenient property that provides a better developer experience.
+
+Take a look at the `user` resource: while the `id` remains a perfectly valid way to get the resource details, the `mail` address is also an unique property that could be used to identify it.
+
+While it is still possible to use the oData filter, such as
+
+`https://graph.microsoft.com/v1.0/users?$filter=mail eq 'bob@contoso.com'`, the returned result is wrapped in an array that needs to be unpacked.
+
+## Solution
+
+---
+
+oData offers resource addressing via an alternate key using the same parentheses-style convention as for the canonical key, with one difference: single-part alternate keys MUST specify the key property name to unambiguously determine the alternate key.
+
+https://graph.microsoft.com/v1.0/users(0) - Retrieves the employee with ID = 0
+https://graph.microsoft.com/v1.0/users(email='bob@contoso.com') Retrieves the employee with the email matching `bob@contoso.com`
+
+## When to Use this Pattern
+
+---
+
+This pattern works and makes sense when the alternate key is good enough to identify a single resource and provides an useful alternative to the client; while it does not work if the resultset has more than one element.
+
+In such case, the workload has two choices:
+
+1. If the workload defines a list of alternate keys for a resource and the requested key is not there, the workload SHOULD return `422`
+2. If the workload does not have a list of alternate keys, it can still query the data source, but if the result yields more than a record, it SHOULD return `422` and encourage the user to use `$filter` rather than returning the first result of the query
+
+## Example
```

Comment:
Sorry I missed this first time around.
I think you need to show the CSDL that is necessary to specify an alternative key (and maybe the primary key for comparison).  I see this kind of thing in the other pattern topics.

Reply:
I have added the necessary CSDL markup in 824b90e